mapfishapp/src/main/webapp/app/addons/notes/js/main.js

```diff
@@ -0,0 +1,149 @@
+Ext.namespace("GEOR.Addons");
+
+GEOR.Addons.Notes = Ext.extend(GEOR.Addons.Base, {
+
+    toggleGroup: "notes",
+
+    /**
+     * Method: init
+     */
+    init: function(record) {
+        if (this.target) {
+            // addon placed in toolbar
+            this.components = this.target.insertButton(this.position, {
+                xtype: "button",
+                enableToggle: true,
+                toggleGroup: this.toggleGroup,
+                tooltip: this.getTooltip(record),
+                icon: GEOR.config.PATHNAME + "/ws/addons/" +
+                    record.get("name").toLowerCase() + "/" +
+                    this.options.icon,
+                listeners: {
+                    "toggle": this.onToggle,
+                    scope: this
+                },
+                scope: this
+            });
+            this.target.doLayout();
+        } else {
+            // addon places in "tools menus"
+            this.item = new Ext.menu.CheckItem({
+                xtype: "button",
+                text: this.getText(record),
+                qtip: this.getQtip(record),
+                icon: GEOR.config.PATHNAME + "/ws/addons/" +
+                    record.get("name").toLowerCase() + "/" +
+                    this.options.icon,
+                checked: false,
+                toggleGroup: this.toggleGroup,
+                listeners: {
+                    "checkchange": this.onToggle,
+                    scope: this
+                }
+            });
+        }
+    },
+    /**
+     * Method: onToggle
+     *
+     */
+    onToggle: function(btn, pressed) {
+        if (pressed) {
+            this.map.events.register("click", this, this.addNote);
+        } else {
+            this.map.events.unregister("click", this, this.addNote);
+        }
+    },
+    /**
+     * Method: addNote
+     *
+     */
+    addNote: function(evt) {
+        var lonlat = this.map.getLonLatFromViewPortPx(evt.xy);
+        lonlat.transform(this.map.projection,
+            new OpenLayers.Projection("EPSG:4326"));
+        this.window = new Ext.Window({
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: ["Ext" is not defined.](https://www.codacy.com/app/georchestra/georchestra/file/2331320306/issues/source?bid=3200181&fileBranchId=3206983#l65)